lib/core/src/lib/form/components/form-field/form-field.component.ts

```diff
@@ -88,14 +92,29 @@ export class FormFieldComponent implements OnInit, OnDestroy {
                     instance.fieldChanged.subscribe((field) => {
                         if (field && this.field.form) {
                             this.visibilityService.refreshVisibility(field.form);
-                            field.form.onFormFieldChanged(field);
+                            this.triggerFormFieldChanged(field);
                         }
                     });
+
+                    if (FormFieldTypes.isReactiveType(instance?.field?.type)) {
```

Comment:
I was expecting this to be called when something has changed into the form field so inside the fieldChanged stream

Reply:
hmmm why? we subscribe to the formRulesEvent when we create this component instance, why do we need to put it under another subscription (nested)? it should listen for these events for the whole lifecycle of the instance

Reply:
if we do not subscribe for formRules changes, we wont be able to update the state of particular instance until the instance.fieldChanged will be triggered (so for example after blur, or input something event).
to make sure I understand you, this is what you mean, right?

```suggestion
                if (componentType) {
                    this.componentRef = this.container.createComponent(componentType);
                    const instance = this.componentRef.instance;
                    instance.field = this.field;
                    instance.fieldChanged.subscribe((field) => {
                        if (field && this.field.form) {
                            this.visibilityService.refreshVisibility(field.form);
                            this.triggerFormFieldChanged(field);

                            if (FormFieldTypes.isReactiveType(instance?.field?.type)) {
                                instance?.updateReactiveFormControl();
                                this.triggerFormFieldChanged(instance.field);
                            }
                        }
                    });
```
(i tested it and it doesnt work)